src/lib.rs

```diff
@@ -1,39 +1,680 @@
+        // ts::Ts { guard: self.guard, object: object }
+        ts::Ts::new(self.guard)
     }
-
 }
+
+pub mod ts {
+
+    extern crate chakracore;
+
+    // declare namespace ts
+    pub struct Ts<'a> {
+        guard: &'a chakracore::context::ContextGuard<'a>,
+        object: chakracore::value::Object,
+    }
+
+    impl<'a> Ts<'a> {
+        pub fn new(guard: &'a chakracore::context::ContextGuard<'a>) -> Ts<'a> {
+            let object = guard.global().get(guard, &chakracore::Property::new(guard, "ts")).into_object().unwrap();
+            Ts { guard: guard, object: object }
+        }
+
+        pub fn version(&self) -> String {
+            self.object.get(self.guard, &chakracore::Property::new(self.guard, "version")).to_string(self.guard)
+        }
+
+        pub fn createSourceFile(&self, fileName: &str, sourceText: &str, languageVersion: ScriptTarget, setParentNodes: Option<bool>, scriptKind: Option<ScriptKind>) -> SourceFileImpl<'a> {
+            println!("do createSourceFile");
+            let function = self.object.get(self.guard, &chakracore::Property::new(self.guard, "createSourceFile")).into_function().unwrap();
+            println!("got function createSourceFile");
+
+            // println!("file");
+            // let file = &chakracore::value::String::new(self.guard, fileName).into();
+            // println!("st");
+            // let st = &chakracore::value::String::new(self.guard, sourceText).into();
+            // println!("lv");
+            // let lv = &chakracore::value::Number::new(self.guard, 5).into();
+            // println!("f");
+            // let f = &chakracore::value::Boolean::new(self.guard, false).into();
+            // println!("sk");
+            // let sk = &chakracore::value::Number::new(self.guard, 3).into();
+            // println!("all params");
+
+            // println!("object: {:?}", self.object.);
+
+            // let rv = function.call_with_this(self.guard, &self.object, &[
+            //     file,
+            //     // st,
+            //     // //  &chakracore::value::Number::new(self.guard, languageVersion).into(), TODO
+            //     // lv,
+            //     // f, // TODO
+            //     // // &chakracore::value::Number::new(self.guard, ScriptKind::TS).into(), // TODO
+            //     // sk,
+            // ]);
+            let rv = function.call_with_this(self.guard, &self.object, &[]);
+            println!("rv: {:?}", rv);
+            
+            let object = rv.unwrap().into_object().unwrap();
+            SourceFileImpl { guard: self.guard, object: object }
+        }
+
+        pub fn createPrinter(&self) -> SourceFileImpl<'a> { // TODO Printer
+            println!("do createPrinter");
+            let function = self.object.get(self.guard, &chakracore::Property::new(self.guard, "createPrinter")).into_function().unwrap();
+            println!("got function createPrinter");
+
+            let rv = function.call_with_this(self.guard, &self.object, &[]);
```

Comment:
@darfink, this quits here with here with:
```
error: process didn't exit successfully: `target\debug\examples\create.exe` (exit code: 255)
```
Any idea what dump mistake I've done? Reproduce with:
```
cargo run --example create
```